### PR TITLE
Bugfix for the issue 'Claims cannot be removed from role'

### LINKED ISSUE #13: Claims cannot be removed from role
There is an issue regardingthe method `RemoveClaimAsync` the following code fails because the list that is enummerated will change during execution. 

```c#
var claims = role.Claims.Where(c => c.ClaimValue == claim.Value).
foreach (var c in claims)
{
    role.Claims.Remove(c);
}
```
Throws
**System.InvalidOperationException**: 'Collection was modified; enumeration operation may not execute.'

I'll try to provide a PR to resolve this

### PR DESCRIPTION
**Pull request by michaelschnyder**
closes #13

### CODE REVIEW

File: test/Maqduni.AspNetCore.Identity.RavenDb.Tests/TestStores.cs [hunk 1]
```diff
@@ -111,6 +111,49 @@ public void RoleCreateAsync(string roleName)
...
+            Assert.True(claims.Any(c => c.Type.Equals(claim.Type, StringComparison.OrdinalIgnoreCase) && c.Value.Equals(claim.Value, StringComparison.OrdinalIgnoreCase)));
+
+            var result = _roleStore.UpdateAsync(role).Result;
+            Assert.True(result.Succeeded);
+        }
+
+        [Theory(DisplayName = "Role WithClaimsRemoveClaimAsync")]
+        [InlineData("Admin", "GraduatedSchoolYear", "2006")]
+        [InlineData("Admin", "GraduatedUniversityYear", "2010")]
+        public void RoleWithClaimsRemoveClaimAsync(string roleName, string claimType, string claimValue)
```
Comment (maqduni) [inline test/Maqduni.AspNetCore.Identity.RavenDb.Tests/TestStores.cs:135]: Please rename the function to RoleClaimsRemoveClaimAsync. A agree that you are implying adding and removing claims, but I think it'd be better to have a unified naming pattern for all functions. I like the idea of adding and removing claims, and I can implement it in all of the tests.

File: test/Maqduni.AspNetCore.Identity.RavenDb.Tests/TestStores.cs [hunk 2]
```diff
@@ -111,6 +111,49 @@ public void RoleCreateAsync(string roleName)
...
+        {
+            var role = _roleStore.FindByNameAsync(roleName).Result;
+            Assert.NotNull(role);
+
+            for (int i = 0; i < 10; i++)
```
Comment (maqduni) [inline test/Maqduni.AspNetCore.Identity.RavenDb.Tests/TestStores.cs:140]: Please just add a single claim, there is no point in adding 10 of them...

**Review by maqduni — CHANGES_REQUESTED** [review]
Please make changes to the unit test. Besides that all looks great! Thanks a lot for your contribution!

File: test/Maqduni.AspNetCore.Identity.RavenDb.Tests/TestStores.cs (hunk 2 above)
Comment (maqduni) [reply test/Maqduni.AspNetCore.Identity.RavenDb.Tests/TestStores.cs:140]: Though it's a good way to point out that claims can be duplicated and by deleting one, all of the dupes will also be gone.

Disregard my first comment, don't make any changes here.

**Review by maqduni — APPROVED** [review]

**Comment by michaelschnyder** [comment]
just got back to work and already merged :) nice. no need for additional work then?

**Comment by maqduni** [comment]
No additional work is required. I figured you worked on it at work and decided to just merge and make the adjustments myself, there were very minor. I published package version 1.3.34, please download and test it. Let me know if you have any questions.

**Pull request merged**